SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DebugInterface/TestDebugInterface.cpp

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestDebugInterface.h"
+#include "SpatialFunctionalTestFlowController.h"
+#include "Utils/DebugInterface.h"
+
+#include "LoadBalancing/LayeredLBStrategy.h"
+#include "LoadBalancing/GridBasedLBStrategy.h"
+
+#include "SpatialGDKFunctionalTests/SpatialGDK/TestActors/ReplicatedTestActorBase.h"
+#include "Kismet/GameplayStatics.h"
+
+ATestDebugInterface::ATestDebugInterface()
+	: Super()
+{
+	Author = "Nicolas";
+	Description = TEXT("Test Debug interface");
+}
+
+namespace
+{
+	FName GetTestTag()
+	{
+		static const FName TestTag(TEXT("TestActorToFollow"));
+		return TestTag;
+	}
+}
+
+void ATestDebugInterface::BeginPlay()
+{
+	Super::BeginPlay();
+
+	AddStep(TEXT("SetupStep"), FWorkerDefinition::AllServers, nullptr, nullptr, [](ASpatialFunctionalTest* NetTest, float DeltaTime)
+	{
+		ATestDebugInterface* Test = Cast<ATestDebugInterface>(NetTest);
+		UWorld* World = Test->GetWorld();
+		Test->DelegationStep = 0;
+		Test->Workers.Empty();
+		Test->bIsOnDefaultLayer = false;
+
+		USpatialGDKDebugInterface::EnableDebugSpatialGDK(World);
```

Comment:
If I understand correctly every test made needs to call this? Not a big fan of it being done manually, also maybe I missed it but this seems to only be accessible in BP?

Reply:
That's how I test it right now. It could be enabled automatically for every test, that's up for discussion.

Reply:
What do you mean by "only accessible to BP" ? The way I setup the interface, it should be accessible from either C++ or BP (not that particular call though, it's a bit too low level to expose in BP IMO).

Reply:
Sorry, meant to write only accessible in C++.

The idea with the framework is that it should allow you to do everything from BP, as a lot of tests are done in BP.

If we don't expose this, then we force people to create an extension of ASpatialFunctionalTest to turn this on, which I'm personally not a fan of. Is there any specific reason why this shouldn't just be active for all tests?

Reply:
I think it would be preferable if it was in the form of a bool on the test actor, so that the test framework can schedule enabling/disabling it itself.
Allowing a test to disable it would be beneficial if we suspect that this debug layer is disrupting the GDK (easier diagnostic).

Reply:
Actually, that would prevent us from using this interface to pin the FlowController actors, so we might need it always enabled in tests anyway :/

Reply:
Any reason we wouldn't move this into `ASpatialFunctionalTest` then?

Reply:
Added a world setting to enable the test interface.